lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -469,6 +469,87 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
+      "Pad", 19, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Torch::ValueTensorType resultType;
+        Value data, pads, constantValue;
+        std::string mode;
+        if (binder.tensorOperandAtIndex(data, 0) ||
+            binder.tensorOperandAtIndex(pads, 1) ||
+            binder.tensorOperandAtIndex(constantValue, 2) ||
+            binder.tensorResultType(resultType) ||
+            binder.customOpNameStringAttr(mode, "mode", "constant"))
+          return failure();
+        Location loc = binder.getLoc();
+
+        auto padsTensorType = pads.getType().cast<Torch::ValueTensorType>();
+        if (!padsTensorType || !padsTensorType.hasSizes()) {
+          return rewriter.notifyMatchFailure(binder.op,
+                                             "Expect non empty pad tensor");
+        }
+        ArrayRef<int64_t> padsShape = padsTensorType.getSizes();
+        int64_t padsRank = padsShape.size();
+        if (padsRank != 1) {
+          return rewriter.notifyMatchFailure(binder.op,
+                                             "Expect 1-D pad tensor");
+        }
+
+        int64_t padsSize = padsShape[0];
+        Value constZero = rewriter.create<Torch::ConstantIntOp>(
+            loc, rewriter.getI64IntegerAttr(0));
+        SmallVector<Value> padsTensorValue;
+        SmallVector<int64_t> emptyShape;
+        Type padsElemType =
+            Torch::ValueTensorType::get(padsTensorType.getContext(), emptyShape,
+                                        padsTensorType.getOptionalDtype());
+        for (unsigned i = 0; i < padsSize; ++i) {
+          Value pos = rewriter.create<Torch::ConstantIntOp>(
+              loc, rewriter.getI64IntegerAttr(i));
+          Value index =
+              Torch::createRank0Tensor(rewriter, loc, padsTensorType, pos);
+          padsTensorValue.emplace_back(
+              rewriter.create<Torch::AtenIndexSelectOp>(loc, padsElemType, pads,
+                                                        constZero, index));
+        }
+        // Rearrange pad tensor to satisfy torch.pad op semantics.
+        SmallVector<Value> padsRearrange;
+        for (unsigned i = 0; i < padsSize / 2; i++) {
+          padsRearrange.emplace_back(padsTensorValue[padsSize / 2 - 1 - i]);
+          padsRearrange.emplace_back(padsTensorValue[padsSize - 1 - i]);
+        }
+
+        Value padsSizeList =
+            rewriter
+                .create<Torch::PrimTolistOp>(
+                    loc,
+                    Torch::ListType::get(rewriter.getType<Torch::IntType>()),
+                    padsRearrange)
+                .getResult(0);
+        Value modeVal = rewriter.create<Torch::ConstantStrOp>(
+            loc, rewriter.getStringAttr(mode));
+
+        auto constValueType =
+            constantValue.getType().cast<Torch::ValueTensorType>();
+        if (!constValueType) {
+          return rewriter.notifyMatchFailure(
+              binder.op, "Expect constant value to be a tensor");
```

Comment:
I think this error message is not correct because the constant value would always be a tensor whenever present. You should instead say that it should be non-none.